machine-runner/README.md

```diff
@@ -225,6 +225,63 @@ Instead, each robot independently ensures that after at five seconds a decision
 Machine runner resolves this conflict by using only the `selected` event that comes first in the Actyx event sort order; in other words, Actyx arbitrarily picks a winner and the losing event is discarded.
 If a robot saw itself winning, started the mission, and then discovers that its win turned out to be invalid, it will have to stop the mission and pick a new one.
 
+## Other Features
+
+### Extra Tags
+
+In the case extra tags are required to be attached in events when invoking commands, these following APIs can be used.
+
+#### On Command Definition
+
+Extra tags can be registered on a command definition. These extra tags will always be attached when the command is invoked.
+
+```typescript
+// State definition
+export const InitialWarehouse = TransportOrderForWarehouse.designState('Initial')
+  .withPayload<{ id: string }>()
+  .command('request', [requested], (ctx, from: string, to: string) => {
+    ctx.withTags([
+      `transport-order-from:${from}`,
+      `transport-order-to:${to}`
+    ])
+    return [{ id: ctx.self.id, from, to }]
+  })
+  .finish()
+
+// Command call
+// The resulting events will include the extra tags
+// `transport-order-from:${from}`,
+// `transport-order-to:${to}`
+const stateAsInitialWarehouse = state
+  .as(InitialWarehouse)?
+  .commands?
+  .request(from: `source`, to: `destination`);
+```
+
+#### On Command Call
+
+Extra tags can be attached when the command is called. This is useful if extra tags are put conditionally and some values that need to be put into the tags are not available within the state-machine.
```

Comment:
> I’d prefer there to be one central place where reviewers should look for the tagging rules instead of having to check the whole code base.

That's a good point. So, scrap this feature?

Reply:
Yes, we can revisit this later.